### PR TITLE
refactor: use dummy inference flow in unit test

### PR DESCRIPTION
**Pull request by ZiniuYu**
This PR replace all mock functions in tasks unit tests with a dummy flow to mimic the behavior of actual inference flow

### CODE REVIEW

File: tests/executor.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+class DummyExecutor(Executor):
+    @requests(on='/caption')
+    def caption(self, docs, **kwargs):
+        for doc in docs:
+            doc.tags['response'] = 'A image of something very nice'
+
+    @requests(on='/encode')
+    def encode(self, docs, **kwargs):
+        for doc in docs:
```
Comment (jemmyshin) [inline tests/executor.py:13]: docs.embeddings = np.random.random((len(docs), 512))

File: tests/test_caption.py [hunk 2]
```diff
@@ -19,29 +17,15 @@
...
-                    tags={'response': 'a image of two rails'},
-                ),
-            ]
-        )
-    ),
-)
-def test_caption_document(inputs):
-    model = Client().get_model('mock-model')
+def test_caption_document(make_flow, inputs):
+    model = BaseClient(
```
Comment (jemmyshin) [inline tests/test_caption.py:21]: can we make this into contest? seems that this model is used for multiple test files

File: tests/executor.py (hunk 1 above)
Comment (ZiniuYu) [reply tests/executor.py:13]: 👌

**Comment by jemmyshin** [comment]
lgtm

**Review by jemmyshin — APPROVED** [review]

**Pull request merged**